src/backend/distributed/commands/utility_hook.c

```diff
@@ -612,6 +658,180 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		 */
 		CitusHasBeenLoaded();
 	}
+
+	if (list_length(ddlJobs) != 0)
+	{
+		return true;
+	}
+
+	if (IsDropSchemaOrTable(parsetree))
+	{
+		/*
+		 * TODO: We could specifically check if it is a:
+		 * - DROP TABLE statement that drops a reference or a citus local table
+		 *   that has a foreign key relationship with a postgres table, or a
+		 * - DROP SCHEMA statement that again drops such a citus table
+		 * but this is more easy and the other approach is much more prune to
```

Comment:
>the other approach is much more prune to
		 * deadlocks. 
		 
 Why is so?

Reply:
For example when dropping a schema, we need to check every citus table in that schema and I think we need to acquire some locks on those citus tables when doing that.
But actually that drop schema/table command could also be a false alarm, so even when we are using local execution, possibility of ending-up with a deadlock would be higher I think.

What do you think ?

Reply:
The session that is executing this piece of the code already acquired very heavy locks on the tables. I don't see any additional potential deadlocks.

Reply:
true, addressed this at 5bae03d

Reply:
anyway removed that code, so resolving